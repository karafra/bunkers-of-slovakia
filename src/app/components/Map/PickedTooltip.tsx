import React from 'react';
import styled from 'styled-components/macro';

export interface PickedTooltipProps {
  x: number;
  y: number;
  objects: {
    name: string;
    description?: string;
  }[];
}

const PickedTooltip = (props: PickedTooltipProps) => {
  return (
    <Wrapper
      style={{
        left: props.x,
        top: props.y,
      }}
      className={"tooltip"}
    >
      {props.objects.map(obj => {
        return (
          <div key={obj.name}>
            <h5>{obj.name}</h5>
            <div>Description: {obj.description || 'unknown'}</div>
          </div>
        );
      })}
    </Wrapper>
  );
};

const Wrapper = styled.div`
  padding: '10px',
  border-radius: '5px',
  color: 'black',
  max-height: '30%',
  background-color: 'whitesmoke',
  position: 'relative',
  min-width: '240px',
  max-width: '10%',
  width: 'fit-content',
  block-size: 'fit-content',
  overflow: 'scroll',
  box-shadow: '0 19px 38px rgba(0,0,0,0.30), 0 15px 12px rgba(0,0,0,0.22)',
  overflow-x: 'hidden',
  pointer-events: 'all',
  z-index: 9,
  font-size: '12px',
`;

export default PickedTooltip;
