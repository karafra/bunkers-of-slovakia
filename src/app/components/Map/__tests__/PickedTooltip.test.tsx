import { render } from '@testing-library/react';
import React from 'react';
import PickedTooltip from '../PickedTooltip';

describe('<PickedTooltip />', () => {
  it('Should render tooltip', () => {
    // Given
    const x = 0;
    const y = 0;
    const name1 = 'name1';
    const name2 = 'name2';
    const description = 'description';
    const objects = [
      {
        name: name1,
        description,
      },
      {
        name: name2,
        description,
      },
    ];
    // When
    const rendering = render(<PickedTooltip x={x} y={y} objects={objects} />);
    // Then
    expect(rendering).toMatchSnapshot();
    expect(
      rendering.container.querySelector('.tooltip')?.querySelectorAll('div')
        .length,
    ).toBe(objects.length * 2);
    expect(rendering.container.querySelector('.tooltip')?.innerHTML).toContain(
      description,
    );
    expect(rendering.container.querySelector('.tooltip')?.innerHTML).toContain(
      name1,
    );
    expect(rendering.container.querySelector('.tooltip')?.innerHTML).toContain(
      name2,
    );
  });
  it('Should render tooltip without', () => {
    // Given
    const x = 0;
    const y = 0;
    const name = 'name';
    const description = 'unknown';
    const objects = [
      {
        name,
      },
    ];
    // When
    const rendering = render(<PickedTooltip x={x} y={y} objects={objects} />);
    // Then
    expect(rendering).toMatchSnapshot();
    expect(
      rendering.container.querySelector('.tooltip')?.querySelectorAll('div')
        .length,
    ).toBe(objects.length * 2);
    expect(rendering.container.querySelector('.tooltip')?.innerHTML).toContain(
      description,
    );
    expect(rendering.container.querySelector('.tooltip')?.innerHTML).toContain(
      name,
    );
  });
});
