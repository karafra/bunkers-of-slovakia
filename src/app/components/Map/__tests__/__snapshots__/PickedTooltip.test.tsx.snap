// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PickedTooltip /> Should render tooltip 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="tooltip interactive"
        style="padding: 10px; border-radius: 5px; color: black; max-height: 30%; background-color: whitesmoke; position: relative; min-width: 240px; max-width: 10%; block-size: fit-content; overflow: scroll; box-shadow: 0 19px 38px rgba(0,0,0,0.30), 0 15px 12px rgba(0,0,0,0.22); overflow-x: hidden; pointer-events: all; z-index: 9; font-size: 12px; left: 0px; top: 0px;"
      >
        <div>
          <h5>
            name1
          </h5>
          <div>
            Description: 
            description
          </div>
        </div>
        <div>
          <h5>
            name2
          </h5>
          <div>
            Description: 
            description
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="tooltip interactive"
      style="padding: 10px; border-radius: 5px; color: black; max-height: 30%; background-color: whitesmoke; position: relative; min-width: 240px; max-width: 10%; block-size: fit-content; overflow: scroll; box-shadow: 0 19px 38px rgba(0,0,0,0.30), 0 15px 12px rgba(0,0,0,0.22); overflow-x: hidden; pointer-events: all; z-index: 9; font-size: 12px; left: 0px; top: 0px;"
    >
      <div>
        <h5>
          name1
        </h5>
        <div>
          Description: 
          description
        </div>
      </div>
      <div>
        <h5>
          name2
        </h5>
        <div>
          Description: 
          description
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`<PickedTooltip /> Should render tooltip without 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="tooltip interactive"
        style="padding: 10px; border-radius: 5px; color: black; max-height: 30%; background-color: whitesmoke; position: relative; min-width: 240px; max-width: 10%; block-size: fit-content; overflow: scroll; box-shadow: 0 19px 38px rgba(0,0,0,0.30), 0 15px 12px rgba(0,0,0,0.22); overflow-x: hidden; pointer-events: all; z-index: 9; font-size: 12px; left: 0px; top: 0px;"
      >
        <div>
          <h5>
            name
          </h5>
          <div>
            Description: 
            unknown
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="tooltip interactive"
      style="padding: 10px; border-radius: 5px; color: black; max-height: 30%; background-color: whitesmoke; position: relative; min-width: 240px; max-width: 10%; block-size: fit-content; overflow: scroll; box-shadow: 0 19px 38px rgba(0,0,0,0.30), 0 15px 12px rgba(0,0,0,0.22); overflow-x: hidden; pointer-events: all; z-index: 9; font-size: 12px; left: 0px; top: 0px;"
    >
      <div>
        <h5>
          name
        </h5>
        <div>
          Description: 
          unknown
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
